crypto/hash/src/main/java/org/apache/shiro/crypto/hash/Md2Hash.java

```diff
@@ -30,10 +30,13 @@
  * techniques and how the overloaded constructors function.
  *
  * @since 0.9
+ * @deprecated and will throw exceptions since 2.0.0, to be removed in 2.1.0.
```

Comment:
Rip anything out in 2.0, removing anything is technically a breaking change. (it's easy to add back in later if/when needed)

Reply:
we can also mark them deprecated now, and then remove all the deprecated bits before the official 2.0 release if you are on the fence about anything.

Another idea, is we could nag users: print a log message: "WARN: Seriously, stop using MD2, here is why <insert some URL here>" 👺

Reply:
+1 for ripping out. We should nag the users in 1.7+ then.
And not just for MD2. For all iteration+salt+pepper based hashes, even if there is no alternative in shiro 1. Seriously.